spec/models/piece_spec.rb

```diff
@@ -58,13 +55,37 @@
 
       expect(piece_one.is_obstructed?(2, 2)).to eq false
     end
+
+    it 'returns true if destination contains piece of the same color(white)' do
```

Comment:
I am still thinking that `is_obstructed?` probably won't be calling the `players_own_piece_is_there?` method later on. I don't think that `is_obstructed?` should rely on any other method . Same with `players_own_piece_is_there?`, they both have one job to do. They could both be called be another method that should incorporate both methods later, like `render_available_moves`.

Reply:
I have no clue on this one.  Personally would say that all the methods should be built up in one place. In this case `piece.rb`. Then just call this final method somewhere else. But this seems like something to ask Sam about because I think it works both ways and there is probably a style choice that needs to be implemented.

Reply:
They would still be in Piece.rb, as would the method that renders available moves. It's just another abstraction of the methods for clarity. Like what you did with the move_to! method.